curator-recipes/src/main/java/org/apache/curator/framework/recipes/nodes/PersistentNode.java

```diff
@@ -445,7 +445,8 @@ private void createNode()
         try
         {
             String existingPath = nodePath.get();
-            String createPath = (existingPath != null && !useProtection) ? existingPath : basePath;
+            String createPath = existingPath == null || (useProtection && !isSequential(mode)) ? basePath
+                    : (useProtection ? basePath + existingPath.substring(existingPath.length()-10) : existingPath);
```

Comment:
I think that this would be cleaner as some if statements. Also, the 10 magic number is not ideal, but I don't believe that we have a constant defining the length of the sequential part of the node name anywhere. Not sure where the best place would be, perhaps just in the CuratorFramework interface. Thoughts @Randgalt ?

Reply:
Thanks @cammckenzie. I reworked this to use if statements and defined a constant for the 10 number. I couldn't find such a constant defined elsewhere; it's hardcoded in the zookeeper code: https://github.com/apache/zookeeper/blob/master/zookeeper-server/src/main/java/org/apache/zookeeper/server/PrepRequestProcessor.java#L662.

I also removed the redundant added static `isSequential(CreateMode)` method, didn't notice that this is already a method on the enum.

Reply:
This might be a good candidate to add to `ZPath.java` - i.e. a method to pull out the sequential number

Reply:
Thanks @Randgalt, I figured that there was a good place to put this, but couldn't remember where it should be. @njhill can you refactor the sequential number extraction into org.apache.curator.utils.ZKPaths class? Or, if you don't have time, let me know and I'll get it done.

Reply:
@Randgalt @cammckenzie sure, I can do that but am wondering about the best semantics for such a utility method. In particular should it first verify that the path ends in 10 digits? (even in the case that it does, I don't think it's guaranteed to be a sequential node). If so and it does not end in 10 digits, what should be the behaviour?

For this scenario in `PersistentNode` we can be "sure" that it's sequential and hence it's probably reasonable (and more efficient) to omit the check. Given that curator has lasted this long without the need for such a method, I'm wondering whether having it is really warranted. Might it be better to wait until N > 1 (if that ever happens)?

Reply:
Good question @njhill  I don't really have a super strong preference, but I'd prefer that the constant defining the size of the sequential bit is not in the PersisteNode class. Perhaps just have a method in ZKPaths that extracts the last 10 digits without any checks. We can make it more defensive down the track if it gets used elsewhere.

I don't think that efficiency is a huge concern here as it's only going to happen when the node is created.

Reply:
@cammckenzie @Randgalt I've now updated it per the suggestions, PTAL!